### PR TITLE
Code refactoring (Client-side && Server-side )

### PR DESCRIPTION
**Pull request by Rebaiahmed**

### CODE REVIEW

**Comment by Rebaiahmed** [comment]
I just make it one PR, If you can merge it

**Comment by gayanvoice** [comment]
Thank you @Rabaiahmed I will merge it by two days.

**Comment by gayanvoice** [comment]
Capture
```
TypeError: "responseJson[0] is undefined"
    renderCountryData Country.js:25
index.js:1375
    e index.js:1375
    renderCountryData Country.js:30

```
The app can not fetches the data from mongoDB

File: client/src/App.js [hunk 1]
```diff
@@ -1,117 +1,31 @@
...
-                      <img alt={"Adem Kouki"} src={"https://avatars0.githubusercontent.com/u/12462188?s=48&v=4"} className={"rounded"}/>
-                  </a>
-
-              </nav>
-              <h6><span className="small">How to contribute? <a
-                  href={"https://github.com/gayanvoice/githubstats/blob/master/CONTRIBUTING.md"}> Add missing countries</a></span>
-              </h6>
-          </div>
-
-          <div className={'align-items-center p-3 mb-4 mt-3 text-center'}>
```
Comment (gayanvoice) [inline client/src/App.js]: You have removed the footer of the app which is used to display that this app githubstats.com is not affiliated with GitHub. I think it should be included in the app.

File: client/src/components/NavbarComponent.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { Link } from 'react-router-dom';
+export function Navbar() {
+  return (
+    <nav className="navbar  navbar-expand-lg  shadow-lg">
+      <div className="navbar-collapse" id="navbarsExampleDefault">
+        <nav className="nav nav-underline">
+          <span className="nav-link">
+            <Link to="/home"><img src={'https://image.flaticon.com/icons/svg/25/25231.svg'} alt={'go to home'} height='24' /></Link>
+          </span>
+          <span className="nav-link">
```
Comment (gayanvoice) [inline client/src/components/NavbarComponent.js]: I put About for just as a reference. Actually there is no need to display in the `Navbar`.

**Comment by gayanvoice** [comment]
file
You have created separate directories for each componenet in the server. Is it necessary? because unlike client there are no multiple files in each directory.

**Comment by Rebaiahmed** [comment]
> file
> You have created separate directories for each component in the server. Is it necessary? because unlike the client there are no multiple files in each directory.

It's a code organization if the application will grow, any new developer needs to reference the necessary files to make his Changement

File: client/src/components/NavbarComponent.js (hunk 2 above)
Comment (Rebaiahmed) [reply client/src/components/NavbarComponent.js]: What' the utility of an empty Page? not linked in the Navbar? useless code

File: client/src/App.js (hunk 1 above)
Comment (Rebaiahmed) [reply client/src/App.js]: I just make the Footer and Navbar as components to make your HTML code more visible !

**Comment by Rebaiahmed** [comment]
> 1. `nodemon `is simply used in the development of the app. But we can not implement `nodemon ` to start the post-build application, because there are no changes that can be done after we deployed. Heroku automatically builds each time when we push the code to its repository.
> 2. `concurrently ` is used to run more than one command at a time. Each time we deploy it to Heroku, it builds the client.
>    https://github.com/gayanvoice/nodejs-graphql/blob/a6bf28d96ecd8b9c471d16281797655c36ecc2f3/server.js#L148
>     This code gets the static build of the client app. So we don't need to create another server to run the React app.

You need to differentiate between the dev mode and prod mode, the folder build doesn't exists  only in the production env, so when I start the project there is no response when I put localhost:port_server, if you are convinced only about your opinion about your code organization that's another point,

**Comment by Rebaiahmed** [comment]
> 1. `nodemon `is simply used in the development of the app. But we can not implement `nodemon ` to start the post-build application, because there are no changes that can be done after we deployed. Heroku automatically builds each time when we push the code to its repository.
> 2. `concurrently ` is used to run more than one command at a time. Each time we deploy it to Heroku, it builds the client.
>    https://github.com/gayanvoice/nodejs-graphql/blob/a6bf28d96ecd8b9c471d16281797655c36ecc2f3/server.js#L148
>     This code gets the static build of the client app. So we don't need to create another server to run the React app.

We need in the package JSON make a start command for dev env and prod mode, Why as a developer should start both projects client-side and server-side each one in a terminal, and I make Changement in teh server-side should always make restart for the app !

**Comment by gayanvoice** [comment]
@Rebaiahmed Accidentally I deleted this branch `#1-refactoring ` I want to merge with into a new branch [https://github.com/gayanvoice/nodejs-graphql/tree/code-refactor-1](https://github.com/gayanvoice/nodejs-graphql/tree/code-refactor-1), but it seems both PR and the branch can not retrieve.

File: client/src/components/NavbarComponent.js (hunk 2 above)
Comment (gayanvoice) [reply client/src/components/NavbarComponent.js]: Yeap, it is a useless code. I used this project to learn to React. Actually, this was my first React app. I used those to study how to React router works with several pages.

**Pull request closed without merging**